azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.toolkits.appservice.service;
+
+import com.microsoft.azure.toolkits.appservice.entity.WebAppEntity;
+import com.microsoft.azure.toolkits.appservice.model.DeployType;
+import com.microsoft.azure.toolkits.appservice.model.PublishingProfile;
+import com.microsoft.azure.toolkits.appservice.model.Runtime;
+import com.microsoft.azure.toolkits.appservice.service.webapp.WebApp;
+
+import java.io.File;
+import java.util.List;
+
+public interface IWebApp {
+    void start();
+
+    void stop();
+
+    void restart();
+
+    void delete();
+
+    void deploy(File file);
+
+    void deploy(DeployType deployType, File file);
+
+    boolean exists();
+
+    WebAppEntity get();
+
+    WebApp.WebAppCreator create();
+
+    WebApp.WebAppUpdater update();
+
+    Runtime getRuntime();
+
+    PublishingProfile getPublishingProfile();
+
+    IWebAppDeploymentSlot deploymentSlot(String slotName);
+
+    List<IWebAppDeploymentSlot> deploymentSlots();
```

Comment:
these method should be invoked by the client with WebAppEnitity.getRuntime()/getPublishingProfile()/... ?
how about remove them in this interface.

Reply:
Yes, but some properties which is widely used in our tools, so may be we need to move some common parameters to interface.